src/safeds/ml/classical/classification/_gradient_boosting.py

```diff
@@ -13,27 +13,34 @@
 
 
 class GradientBoosting(Classifier):
-    """Gradient boosting classification.
+    """
+    Gradient boosting classification.
 
     Parameters
     ----------
     learning_rate : float
         The larger the value, the more the model is influenced by each additional tree. If the learning rate is too
         low, the model might underfit. If the learning rate is too high, the model might overfit.
+    number_of_trees: int
+        The number of boosting stages to perform. Gradient boosting is fairly robust to over-fitting so a large
+        number usually results on better performance.
 
     Raises
     ------
     ValueError
-        If `learning_rate` is non-positive.
+        If `learning_rate` is non-positive or the `number_of_trees` is less than or equal to 0.
     """
 
-    def __init__(self, learning_rate: float = 0.1) -> None:
+    def __init__(self, learning_rate: float = 0.1, number_of_trees: int = 100) -> None:
```

Comment:
I think the number of trees is more essential than the learning rate, so the order of parameters should be switched. 

See also the distinction between _model hyperparameters_ and _algorithm hyperparameters_:

> Hyperparameters can be classified as **model hyperparameters**, that cannot be inferred while [fitting the machine to the training set](https://en.wikipedia.org/wiki/Model_fitting) because they refer to the [model selection](https://en.wikipedia.org/wiki/Model_selection) task, or **algorithm hyperparameters**, that in principle have no influence on the performance of the model but affect the speed and quality of the learning process. An example of a model hyperparameter is the topology and size of a neural network. Examples of algorithm hyperparameters are [learning rate](https://en.wikipedia.org/wiki/Learning_rate) and batch size as well as mini-batch size. Batch size can refer to the full data sample where mini-batch size would be a smaller sample set.

[Source](https://en.wikipedia.org/wiki/Hyperparameter_(machine_learning))

Maybe we should add a general [guideline](https://stdlib.safeds.com/en/latest/development/guidelines/), that _model hyperparameter_ should be listed before _algorithm hyperparameters_.